--- asterisk-1.2.4/apps/app_dial.c	2006-01-25 02:50:52.000000000 +0100
+++ asterisk-1.2.4-answer/apps/app_dial.c	2006-02-13 02:59:17.000000000 +0100
@@ -554,10 +554,13 @@
 									       OPT_CALLEE_HANGUP | OPT_CALLER_HANGUP |
 									       OPT_CALLEE_MONITOR | OPT_CALLER_MONITOR |
 									       DIAL_NOFORWARDHTML);
+								ast_indicate(in, AST_CONTROL_ANSWER);
+
 							}
 							/* If call has been answered, then the eventual hangup is likely to be normal hangup */
 							in->hangupcause = AST_CAUSE_NORMAL_CLEARING;
 							o->chan->hangupcause = AST_CAUSE_NORMAL_CLEARING;
+
 							break;
 						case AST_CONTROL_BUSY:
 							if (option_verbose > 2)
